test/functional/options/defaults_spec.lua

```diff
@@ -328,3 +329,64 @@ describe('XDG-based defaults', function()
     end)
   end)
 end)
+
+
+describe('stdpath()', function()
+  before_each(function()
+    clear({env={
+      XDG_CONFIG_HOME='cat',
+      XDG_CONFIG_DIRS='siamese:persian',
+      XDG_DATA_HOME='dog',
+      XDG_DATA_DIRS='chihuahua:terrier',
+      XDG_CACHE_HOME='capybara',
+    }})
+  end)
+
+  it('knows the config home', function()
+    eq('cat/nvim', eval('stdpath("config")'))
+  end)
+
+  it('knows the data home', function()
+    if os_name() == 'windows' then
+      eq('dog\\nvim-data', eval('stdpath("data")'))
+    else
+      eq('dog/nvim', eval('stdpath("data")'))
+    end
+  end)
+
+  it('knows the cache home', function()
+    if os_name() == 'windows' then
+      eq('capybara\\nvim', eval('stdpath("cache")'))
+    else
+      eq('capybara/nvim', eval('stdpath("cache")'))
+    end
+  end)
+
+  it('knows the config dirs', function()
+    local gotten = eval('stdpath("config_dirs")')
+    if os_name() == 'windows' then
+      eq('siamese\\nvim', gotten[1])
+      eq('persian\\nvim', gotten[2])
+    else
+      eq('siamese/nvim', gotten[1])
+      eq('persian/nvim', gotten[2])
+    end
+    eq(2, #gotten)
+  end)
+
+  it('knows the data dirs', function()
+    local gotten = eval('stdpath("data_dirs")')
+    if os_name() == 'windows' then
+      eq('chihuahua\\nvim', gotten[1])
+      eq('terrier\\nvim', gotten[2])
+    else
+      eq('chihuahua/nvim', gotten[1])
+      eq('terrier/nvim', gotten[2])
+    end
+    eq(2, #gotten)
+  end)
+
+  it('accepts unknown strings', function()
+    eq('', eval('stdpath("cavey")'))
+  end)
```

Comment:
But that isn't a function of `stdpath()`; [`stdpaths_get_xdg_var()`](https://github.com/neovim/neovim/blob/15af803cce0221588011879723ccfc2ade51ea98/src/nvim/os/stdpaths.c#L54-L81) takes care of that for `stdpath()`.

I don't see any tests for `stdpaths_get_xdg_var()` but I may be missing them.